### DISCUSSION TITLE
Clarifications about setting LIB environment variable on Windows

### DISCUSSION THREAD

**Discussion by GyulyVGC** [rust-pcap/pcap#301] [Q&A]
Hey guys I have a doubt which keeps popping back in my mind.

I know there is the need to set the LIB env var to a folder of the Npcap SDK in order to **_build_**.
But I'm wondering if this process is also needed to just **_execute_**.

If I build an application using pcap, and later share it for someone else to run, who receives the app needs to set the LIB variable as well?
I have tried removing LIB after a build and the software keeps running, so I guess it's not needed.
But still I'm not sure, because some people are reporting problems under some circustances (and I cannot understand if it could be related to the LIB var or if it has nothing to do).

Could you kindly explain how is the SDK used? Is it used only during the build process?

Thanks in advance.

**Comment by rust1248** [comment]
The user will have to install the library on their system, but I think you can compile statically to include the lib in your binary.

**Reply by GyulyVGC** [reply]
So I guess the SDK is automatically included in the binary once it's built. Is that correct?

**Reply by Wojtek242** [reply]
Not by default. 

By default libpcap/npcap is linked against dynamically. The `LIBPCAP_LIBDIR` is a variable that tells the build system (i.e. cargo) where to find the library to link against, but tells pcap nothing about where to find it once the binary is run. Once compiled, the binary must be able to find the library using standard OS-specific conventions. On Linux these libraries are either in a standard place (as installed by the package manager) or you can specify additional locations with `LD_LIBRARY_PATH`. I do not know what the Windows equivalent is.

Or, as @rust1248 pointed out, you can compile libpcap/npcap into your binary statically which means the library will be included with the binary. This requires libpcap to be present and possible to find during compilation, but there no longer is such a requirement during runtime.

**Reply by GyulyVGC** [reply]
My doubts were more with respect to Windows.
So, in practical terms: is the SDK (on Windows) required to _run_ a binary built with cargo? Npcap itself is required for sure, but is it also required the SDK in Windows, to _run_? 
I hope to have explained myself better.

**Reply by Wojtek242** [reply]
That depends what Npcap means by SDK. If it means headers only then I would say no, you don't need the SDK to run the binary. I don't use Windows myself so I can't say conclusively.

**Reply by GyulyVGC** [reply]
Thanks anyway for the support, I will update you if I discover something more!
